7_react/react1/week3/homework/todo-list/src/components/TodoItem.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+
+const TodoItem = React.memo(({ todo, toggleTodo, updateTodo, deleteTodo }) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newDescription, setNewDescription] = useState(todo.description);
+
+  const handleDescriptionChange = (e) => {
+    setNewDescription(e.target.value);
+  };
+
+  const handleUpdate = () => {
+    if (newDescription.trim()) {
+      if (todo.deadline && new Date(todo.deadline) < new Date()) {
```

Comment:
I do not think it is necessary to check the `deadline` if you want to change just the description and not the `deadline` itself, also right now if you try to edit an element with the deadline in past, there is no other choice than delete the element otherwise you got stuck

Reply:
Also I think you were supposed to update all todo's props `deadline` included